stages/org.osbuild.coreos.platforms

```diff
@@ -0,0 +1,159 @@
+        raise FileNotFoundError(f"Source file '{source_path}' does not exist.")
+    try:
+        # Copy the file to the destination
+        shutil.copy2(source_path, destination_path)
+        print(f"File '{source_path}' copied to '{destination_path}' successfully.")
+    except Exception as e:
+        print(f"Error occurred while copying file: {e}")
+
+
+def append_options_to_file(file_path, opts, key, bls=None):
+    if os.path.isfile(file_path):
+        # Read the file and then append to key
+        with open(file_path, encoding="utf8") as f:
+            lines = f.read().splitlines()
+        with open(file_path, "w", encoding="utf8") as f:
+            for line in lines:
+                if line.startswith(key):
+                    if bls is not None:
+                        f.write(f"{line} {' '.join(opts)}\n")
+                    else:
+                        f.write(f"{line}\n")
+                        for opt in opts:
+                            f.write(f"{opt}\n")
+                else:
+                    f.write(f"{line}\n")
+    else:
+        print(f"File not found: {file_path}")
+
+
+def process_json(json_file_path, platform, key):
+    try:
+        with open(json_file_path, 'r') as file:
+            data = json.load(file)
+            if platform in data and key in data[platform]:
+                return data[platform][key]
+    except Exception as e:
+        raise Exception(f"An error occurred: {str(e)}")
+    return None
+
+
+def main(paths, tree, options):
+
+    bootpath = options.get("bootpath", "tree:///boot")
+    deployment = options.get("deployment")
+    platforms_json =  options.get("platforms_json")
+    platform = options.get("platform")
+
+    kopts = []
+    kopts.append(f"ignition.platform.id={platform}")
+    print(kopts)
+    url = urlparse(bootpath)
+    scheme = url.scheme
+    if scheme == "tree":
+        root = tree
+    elif scheme == "mount":
+        root = paths["mounts"]
+    else:
+        raise ValueError(f"Unsupported scheme '{scheme}'")
+    assert url.path.startswith("/")
+    bootroot = root + url.path
+
+    osname, ref, serial = ostree.parse_deployment_option(root, deployment)
+    root = ostree.deployment_path(root, osname, ref, serial)
+    platforms_source_path = f"{root}/{platforms_json}"
+    platforms_dest_path = f"{bootroot}/coreos/platforms.json"
+
+    # Copy platforms.json to the boot partition
+    copy_file(platforms_source_path, platforms_dest_path)
+    # Get kernel arguments if it exits, it may not
+    kopt = process_json(platforms_dest_path, platform, "kernel_arguments")
+    if kopt is not None:
+         kopts.append(kopt)
+    # Add kernel arguments to bls config
+    # There is unlikely to be more than one bls config, but just
+    # in case we'll iterate over them.
+    entries = []
+    for entry in glob.glob(f"{bootroot}/loader/entries/*.conf"):
+        append_options_to_file(entry, kopts, 'options ', True)
+        entries.append(entry)
```

Comment:
Feels a bit like we should take this code and the code in the `bls-append` stage and hoist it into a shared location and use it in both places rather than copy and pasting?

Reply:
ahh I see.. you tried to make the function generic so we could use it for appending bls or appending the grub console options.. even still might be nice to have that more generic piece in a shared place and then use it in both bls-append and here?